### PR TITLE
`KMS`:  add `google_kms_crypto_key_latest_version` data source

### PR DESCRIPTION
**Pull request by BBBmau**
Continuation of https://github.com/GoogleCloudPlatform/magic-modules/pull/11455

Second step where we focus on grabbing the latest crypto key version (singular data source)

This PR depends on the referenced PR, plural data source must be merged before this can be properly reviewed.

The only major difference is once the list of cryptoKeyVersions is set we grab the latest one which is done in this line here:
https://github.com/GoogleCloudPlatform/magic-modules/pull/11456/files#diff-7c2dc734dd72075cca3912ca10f4c3a3736ea934186f1392fc7e0764a37eac53R99-R103

filter is added in since it gives the user the ability to grab the latest in either the enabled state or disabled state.

PR also includes some clean up of the plural `cryptoKeyVersions` data source such as adding in the missing filter argument in the docs.

Test passing:
```hcl
└─(13:33:59 on main ✹ ✭)──> envchain GCLOUD make testacc TEST=./google/services/kms TESTARGS='-run TestAccDataSourceGoogleKmsCryptoKeyLatestVersion_basic'                    ──(Fri,Aug23)─┘
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/kms -v -run TestAccDataSourceGoogleKmsCryptoKeyLatestVersion_basic -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceGoogleKmsCryptoKeyLatestVersion_basic
2024/08/23 13:36:12 [INFO] Authenticating using configured Google JSON 'credentials'...
2024/08/23 13:36:12 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]
2024/08/23 13:36:12 [INFO] Authenticating using configured Google JSON 'credentials'...
2024/08/23 13:36:12 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]
2024/08/23 13:36:12 [DEBUG] Waiting for state to become: [success]
2024/08/23 13:36:13 [INFO] Terraform is using this identity: mauricio-alvarezleon@hc-terraform-testing.iam.gserviceaccount.com
2024/08/23 13:36:13 [INFO] Instantiating Google Cloud KMS client for path https://cloudkms.googleapis.com/
2024/08/23 13:36:13 [DEBUG] Retry Transport: starting RoundTrip retry loop
2024/08/23 13:36:13 [DEBUG] Retry Transport: request attempt 0
2024/08/23 13:36:13 [DEBUG] Retry Transport: Stopping retries, last request was successful
2024/08/23 13:36:13 [DEBUG] Retry Transport: Returning after 1 attempts
2024/08/23 13:36:13 [DEBUG] Retry Transport: starting RoundTrip retry loop
2024/08/23 13:36:13 [DEBUG] Retry Transport: request attempt 0
2024/08/23 13:36:13 [DEBUG] Retry Transport: Stopping retries, last request was successful
2024/08/23 13:36:13 [DEBUG] Retry Transport: Returning after 1 attempts
--- PASS: TestAccDataSourceGoogleKmsCryptoKeyLatestVersion_basic (14.10s)
PASS
ok      github.com/hashicorp/terraform-provider-google/google/services/kms      15.502s
```

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_kms_crypto_key_latest_version`
```

### CODE REVIEW

**Comment by BBBmau** [comment]
Marking as ready for review after including the docs, I'm requesting specifically @slevenick due to already reviewing the plural data source. There's not much to review from this PR since it depends on the logic of plural `cryptoKeyVersions` data source to get the latest `cryptoKeyVersion`

File: mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_latest_version.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+				Computed: true,
+			},
+			"filter": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: `
+					The filter argument is used to add a filter query parameter that limits which cryptoKeyVersions are retrieved by the data source: ?filter={{filter}}.
+					Example values:
+					
+					* "name:my-cryptokey-version-" will retrieve cryptoKeyVersions that contain "my-key-" anywhere in their name. Note: names take the form projects/{{project}}/locations/{{location}}/keyRings/{{keyRing}}/cryptoKeys/{{cryptoKey}}/cryptoKeyVersions/{{cryptoKeyVersion}}.
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_latest_version.go:49]: my-key- vs my-cryptokey-version- don't match

**Review by slevenick — APPROVED** [review]

File: mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_latest_version.go (hunk 1 above)
Comment (BBBmau) [reply mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_latest_version.go]: Thanks for catching this, i made the description match what's in the `latest_version.html.markdown` file.

**Comment by BBBmau** [comment]
@slevenick thanks again for reviewing both PRs! 😄

**Pull request merged**